cli/src/semgrep/semgrep_main.py

```diff
@@ -197,24 +198,27 @@ def run_rules(
                 (
                     dep_rule_matches,
                     dep_rule_errors,
-                    targeted_lockfiles,
                 ) = generate_unreachable_sca_findings(
                     rule, target_manager, already_reachable
                 )
                 rule_matches_by_rule[rule].extend(dep_rule_matches)
                 output_handler.handle_semgrep_errors(dep_rule_errors)
-                output_extra.all_targets.union(targeted_lockfiles)
             else:
                 (
                     dep_rule_matches,
                     dep_rule_errors,
-                    targeted_lockfiles,
                 ) = generate_unreachable_sca_findings(
                     rule, target_manager, lambda p, d: False
                 )
                 rule_matches_by_rule[rule] = dep_rule_matches
                 output_handler.handle_semgrep_errors(dep_rule_errors)
-                output_extra.all_targets.union(targeted_lockfiles)
+
+        # Generate stats per lockfile:
+        for ecosystem in ECOSYSTEM_TO_LOCKFILES.keys():
+            for lockfile in target_manager.get_lockfiles(ecosystem):
+                # Add lockfiles as a target that was scanned
+                output_extra.all_targets.add(lockfile)
```

Comment:
Technically this is an approximation of the truth instead of the exact truth

Reply:
Could be convinced to revert this part to old behavior

Reply:
yeah (2) is a good reason to keep old way this was calculated

Reply:
We switched over to always respecting semgrepignore. I moved the lockfile finding logic used in reachability rules into TargetManager so this should be an accurate depiction of true scanned files now